Source/WebCore/rendering/RenderLayerCompositor.h

```diff
@@ -100,6 +100,12 @@ enum class ScrollCoordinationRole {
     Positioning         = 1 << 5,
 };
 
+enum class ViewportConstrainedSublayers : uint8_t {
+    NoLayers,
+    AnchorLayer,
+    ClippingAndAnchorLayers,
+};
```

Comment:
Optional: does the `Layer[s]` prefix in the case names help with readability? If not, maybe just:

```suggestion
enum class ViewportConstrainedSublayers : uint8_t {
    None,
    Anchor,
    ClippingAndAnchor,
};
```

Reply:
Makes sense!